lib/routes/account.js

```diff
@@ -1251,49 +1404,116 @@ module.exports = function (
+                push.notifyUpdate(accountResetToken.uid, 'passwordReset')
+
+                return db.account(accountResetToken.uid)
+              }
+            )
+            .then(
+              function (accountData) {
+                account = accountData
+                log.activityEvent('account.reset', request, {
+                  uid: account.uid.toString('hex')
+                })
+                log.event(
+                  'reset',
+                  {
+                    uid: account.uid.toString('hex') + '@' + config.domain,
+                    generation: account.verifierSetAt
+                  }
+                )
+                return customs.reset(account.email)
+              }
+            )
+            .then(
+              function () {
+                return password.unwrap(account.wrapWrapKb)
+              }
+            )
+            .then(
+              function (wrapKbData) {
+                wrapKb = wrapKbData
+              }
+            )
+        }
+
+        function createSessionToken () {
+          // Create a verified sessionToken
+          var sessionTokenOptions = {
+            uid: account.uid,
+            email: account.email,
+            emailCode: account.emailCode,
+            emailVerified: account.emailVerified,
+            verifierSetAt: account.verifierSetAt
+          }
+
+          return db.createSessionToken(sessionTokenOptions, request.headers['user-agent'])
+            .then(
+              function (result) {
+                sessionToken = result
+              }
+            )
+        }
+
+        function createKeyFetchToken () {
+          if (wantsKeys(request)) {
+            return db.createKeyFetchToken({
+                uid: account.uid,
+                kA: account.kA,
+                wrapKb: wrapKb,
+                emailVerified: account.emailVerified
               })
-              log.event(
-                'reset',
-                {
-                  uid: account.uid.toString('hex') + '@' + config.domain,
-                  generation: account.verifierSetAt
+              .then(
+                function (result) {
+                  keyFetchToken = result
                 }
               )
-              return customs.reset(account.email)
-            }
-          )
-          .then(
-            function () {
-              return {}
-            }
-          )
-          .done(reply, reply)
+          }
+        }
+
+        function createResponse () {
+          var response = {
+            uid: sessionToken.uid.toString('hex'),
+            sessionToken: sessionToken.data.toString('hex'),
+            verified: sessionToken.emailVerified,
+            authAt: sessionToken.lastAuthAt()
+          }
+
+          if (wantsKeys(request)) {
+            response.keyFetchToken = keyFetchToken.data.toString('hex')
+            response.unwrapBKey = (butil.xorBuffers(account.kA, account.wrapWrapKb)).toString('hex')
```

Comment:
@vbudhram - can you remove the `unwrapBKey` from the response? This was a bad request on my part, generating `unwrapBKey` is now handled by the fxa-js-client.